codegenerator/cli/templates/dynamic/codegen/src/db/Entities.res.hbs

```diff
@@ -23,6 +23,240 @@ let isIndex = true
 @genType
 type whereOperations<'entity, 'fieldType> = {eq: 'fieldType => promise<array<'entity>>}
 
+module EntityHistory = {
+  type historyFieldsGeneral<'a> = {
+    chain_id: 'a,
+    block_timestamp: 'a,
+    block_number: 'a,
+    log_index: 'a,
+  }
+
+  type historyFields = historyFieldsGeneral<int>
+
+  type historyRow<'entity> = {
+    current: historyFields,
+    previous: option<historyFields>,
+    entityData: 'entity,
+  }
+
+  type previousHistoryFields = historyFieldsGeneral<option<int>>
+
+  let previousHistoryFieldsSchema = S.object(s => {
+    chain_id: s.field("previous_entity_history_chain_id", S.null(S.int)),
+    block_timestamp: s.field("previous_entity_history_block_timestamp", S.null(S.int)),
+    block_number: s.field("previous_entity_history_block_number", S.null(S.int)),
+    log_index: s.field("previous_entity_history_log_index", S.null(S.int)),
+  })
+
+  let currentHistoryFieldsSchema = S.object(s => {
+    chain_id: s.field("entity_history_chain_id", S.int),
+    block_timestamp: s.field("entity_history_block_timestamp", S.int),
+    block_number: s.field("entity_history_block_number", S.int),
+    log_index: s.field("entity_history_log_index", S.int),
+  })
+
+  let makeHistoryRowSchema: S.t<'entity> => S.t<historyRow<'entity>> = entitySchema =>
+    S.object(s => {
+      {
+        "current": s.flatten(currentHistoryFieldsSchema),
+        "previous": s.flatten(previousHistoryFieldsSchema),
+        "entityData": s.flatten(entitySchema),
+      }
```

Comment:
Using flatten here with a transform below to the actual record was the best way I could think of creating this schema.